src/main.ts

```diff
@@ -205,17 +207,13 @@ const exec = (
   }
 
   const preString = rows.join('\n');
-  const pre = preString[Symbol.iterator]();
-  let currentPre = pre.next();
-  let nextPre = pre.next();
-
-  // We need to keep a separate index as `String#slice()` works on Unicode code units, while `pre` is an array of codepoints.
   let preStringIndex = 0;
 
-  while (!currentPre.done) {
-    const character = currentPre.value;
-    const nextCharacter = nextPre.value;
-
+  while (preStringIndex < preString.length) {
+    let character = preString[preStringIndex];
+    if (character >= '\ud800' && character <= '\udbff') {
+      character = String.fromCodePoint(preString.codePointAt(preStringIndex) ?? 0);
```

Comment:
when can `codePointAt` be nullish?

maybe we should just do this:

```ts
if (character >= '\ud800' && character <= '\udbff') {
  const characterCodePoint = preString.codePointAt(preStringIndex);
  if (characterCodePoint) {
    character = String.fromCodePoint(characterCodePoint);
  }
}
```

to avoid appending a bull byte somehow